### DISCUSSION TITLE
Cool idea, but seems to suffer from a lack of good prompt engineering?

### DISCUSSION THREAD

**Discussion by Jawabiscuit** [ItzCrazyKns/Vane#138] [General]
I role-played as a "normie" for my first test query and search gave me the wrong answer. Now I distrust every successive answer by default. This is an unfortunate problem for most, if not all, AI tools.

I have had an OpenAI subscription for quite some time and have access to gpt4 turbo and o models. Did I miss something or is there no config for which model is being used?

![chrome_W9s8pwEeMH](https://github.com/ItzCrazyKns/Perplexica/assets/219464/a2fbe89a-c820-4523-a167-23bebeb5c222)
![o2Ng3ZcP6h](https://github.com/ItzCrazyKns/Perplexica/assets/219464/041f3292-588d-4d59-8185-50b40a6ad884)

As a technical individual, I'm digging the tech stack being used and the ease of self-hosting with docker though. Very slick.

**Comment by ItzCrazyKns** [comment]
> I role-played as a "normie" for my first test query and search gave me the wrong answer. Now I distrust every successive answer by default. This is an unfortunate problem for most, if not all, AI tools.
> 
> I have had an OpenAI subscription for quite some time and have access to gpt4 turbo and o models. Did I miss something or is there no config for which model is being used?
> 
> ![chrome_W9s8pwEeMH](https://private-user-images.githubusercontent.com/219464/333655791-a2fbe89a-c820-4523-a167-23bebeb5c222.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY1Njg0NjMsIm5iZiI6MTcxNjU2ODE2MywicGF0aCI6Ii8yMTk0NjQvMzMzNjU1NzkxLWEyZmJlODlhLWM4MjAtNDUyMy1hMTY3LTIzYmViZWI1YzIyMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyNFQxNjI5MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMDJhNzNiODZkZTI4NjRmMzAzMTEzMWU4YmJiMzk5YzA3NzkxZGZiNDRiNTA3YWMzMTI4ODM2ODYyZDI0ZWNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Uo32Q2Q5QLDVQWe4vdiSzkQnBKgoZ3iqr5y2e8lzDE8) ![o2Ng3ZcP6h](https://private-user-images.githubusercontent.com/219464/333655822-041f3292-588d-4d59-8185-50b40a6ad884.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY1Njg0NjMsIm5iZiI6MTcxNjU2ODE2MywicGF0aCI6Ii8yMTk0NjQvMzMzNjU1ODIyLTA0MWYzMjkyLTU4OGQtNGQ1OS04MTg1LTUwYjQwYTZhZDg4NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyNFQxNjI5MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NjU2ZDcxNWIzNjdjZGE1ZDc1OGIzMTU1ZWFiNDJmZGE3NzI1YzNkOGRkNGRhNmUwNjEwZDRiODgwZWFhOTY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Boug0QpzgYbVzeczZibyujso45jnZtf9MiiM84Cgqmk)
> 
> As a technical individual, I'm digging the tech stack being used and the ease of self-hosting with docker though. Very slick.

Hi, the query is first passed to an agent that **understands** the question or the query and decides wether there is a need to search the web or not. Sometimes it won't work perfectly and no context is passed to the answer generation chain which then hallucinates the answer. At the end of the day it just predicts stuff but we'll eventually fix it.

**Reply by TomLucidor** [reply]
Maybe there are AutoPrompt tools that takes human feedback and guess how to enhance prompting? Not sure how that would look in multi-hop information + splinternet happening.

**Comment by Zirgite** [comment]
This was in the past. Different models react differently. Also the recent version the prompts are updated. I use custom prompts though.
My verdict is that it is better than perplexity vanilla but below perplexity pro. Especially perplexity with the deep seek reasoning is a monster way better than everything until now.

Response 06.02.25
## OpenAI's Latest Model Release: o3-mini

OpenAI has recently unveiled a new addition to its AI model series, the o3-mini. This launch comes on the heels of the DeepSeek R1 release, which made waves in the AI community just weeks prior.[1][2]

### The o3-Mini Model
The o3-mini is designed with efficiency and cost-effectiveness in mind.[1] It belongs to OpenAI's "reasoning" model series,[1][11] a family of models that prioritize thoughtful responses[4] and complex task reasoning.[3] This approach sets it apart from traditional AI models, which often focus on speed rather than thoroughness.

### Key Features
- **Cost-Efficient:** The o3-mini is intended to be more affordable compared to other OpenAI models,[1][2] making advanced AI capabilities more accessible.
- **Reasoning Capabilities:** This model excels in open-ended reasoning tasks, a capability that has been lacking in previous generations of AI.[9]
- **Competitive Edge:** With the surprise success of DeepSeek's rival product, OpenAI aims to maintain its position at the forefront of AI innovation with this timely release.[2][8]

### Availability and Access
The o3-mini is currently available for ChatGPT Enterprise and Edu users,[10] while developers can access it through the API with certain usage limits in place.[10] OpenAI plans to increase these limits after additional testing.[10]

## Conclusion
OpenAI's latest model, o3-mini, demonstrates the company's commitment to pushing the boundaries of AI technology. By focusing on reasoning capabilities and cost efficiency, OpenAI aims to make advanced AI more accessible and competitive in an increasingly crowded market.

Citations:
[1] https://openai.com/index/openai-o3-mini/
[2] https://www.technologyreview.com/2025/01/31/1110757/openai-makes-its-reasoning-model-for-free/
[3] https://help.openai.com/en/articles/9624314-model-release-notes
[4] https://techcrunch.com/2024/12/20/openai-announces-new-o3-model/
[5] https://platform.openai.com/docs/models
[6] https://learn.microsoft.com/en-us/azure/ai-services/openai/whats-new
[7] https://openai.com/o1/
[8] https://www.theguardian.com/business/2025/jan/31/openai-to-release-new-artificial-intelligence-model-for-free
[9] https://www.technologyreview.com/2024/09/17/1104004/why-openais-new-model-is-such-a-big-deal/
[10] https://openai.com/index/introducing-openai-o1-preview/
[11] https://techcrunch.com/2025/01/31/openai-launches-o3-mini-its-latest-reasoning-model/
[12] https://www.cnn.com/2024/05/13/tech/openai-altman-new-ai-model-gpt-4o/index.html
[13] https://www.datacamp.com/blog/o3-openai
[14] https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models
[15] https://www.wired.com/story/openai-o3-mini-release/
